### PR TITLE
chore(isConformant): add default and onChange tests for autocontrolled props

### LINKED ISSUE #2119: Components: add onChange-like callback
## Bug Report
Some components do not have the proper `onChange` type callbacks. For example `Menu` does not have `onActiveIndexChange`. Also Tree etc.

### Expected Result
Components should have these callbacks for the controlled props.

### PR DESCRIPTION
**Pull request by silviuaavram**
Fixes https://github.com/microsoft/fluent-ui-react/issues/2119

Check for each autocontrolled prop:
- there is an `on${capitalized_prop}Change` or, if autocontrolled prop is `value`, then check `onChange`.
- there is a `default{capitalized_prop}`.

### CODE REVIEW

File: packages/react/test/specs/commonTests/isConformant.tsx [hunk 1]
```diff
@@ -308,6 +308,33 @@ export default function isConformant(
         handledProps: expectedProps,
       })
     })
+
+    test('autoControlled props should have onChange props associated', () => {
```
Comment (mnajdova) [inline packages/react/test/specs/commonTests/isConformant.tsx]: I am all for these changes, it will ensure consistency :) 🚀

Comment (mnajdova) [reply packages/react/test/specs/commonTests/isConformant.tsx]: @joschect FYI this is a nice conformant test for the autocontrolled props

File: docs/List.info.json [hunk 2]
```diff
@@ -0,0 +1,282 @@
+{
```
Comment (layershifter) [inline docs/List.info.json]: Cleanup files there please...

File: packages/react/test/specs/commonTests/isConformant.tsx [hunk 3]
```diff
@@ -308,6 +311,32 @@ export default function isConformant(
...
+              1,
+            )}Change`
+
+          expect(Component.handledProps).toContain(expectedOnChangeProp)
+        })
+      }
+    })
+
+    test('autoControlled props should have default props', () => {
+      if (Component.autoControlledProps && Component.autoControlledProps.length) {
```
Comment (layershifter) [inline packages/react/test/specs/commonTests/isConformant.tsx]: In functional components we don't have `autoControlled` props anymore...

File: docs/List.info.json (hunk 2 above)
Comment (silviuaavram) [reply docs/List.info.json]: If these are not supposed to be here then we should add them to gitignore.

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -813,7 +829,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
   }
 
   handleSearchQueryChange = (searchQuery: string) => {
-    this.setStateAndInvokeHandler('onSearchQueryChange', null, {
+    this.setStateAndInvokeHandler(['onSearchQueryChange', 'onHighlightedIndexChange'], null, {
```
Comment (silviuaavram) [inline packages/react/src/components/Dropdown/Dropdown.tsx:832]: utility function extended to call multiple handlers.

File: CHANGELOG.md [hunk 5]
```diff
@@ -23,8 +23,9 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
 - Returned function from `useAccessibility` no longer keeps the same reference @layershifter ([#2268](https://github.com/microsoft/fluent-ui-react/pull/2268))
-- Changed `avatarBorderWidth` and `statusBorderWidth` avatar variables types from number to string and updated styles in Teams theme @mnajdova ([#2238](https://github.com/microsoft/fluent-ui-react/pull/2238))
 - Add logic for mounting/removing elements when they are shown/hidden using the `Animation` component; `Animation` component is not rendering element anymore, just applying classes to it's children @mnajdova ([#2115](https://github.com/microsoft/fluent-ui-react/pull/2115))
+- Restricted prop set in the `Button`, `Avatar`, `Box` and `Image` styles; changed `avatarBorderWidth` and `statusBorderWidth` avatar variables types from number to string and updated styles in Teams theme @mnajdova ([#2238](https://github.com/microsoft/fluent-ui-react/pull/2238))
+- Restricted prop set in the `List` & `ListItem` @layershifter ([#2238](https://github.com/microsoft/fluent-ui-react/pull/2238))
```
Comment (mnajdova) [inline CHANGELOG.md:28]: Please update your branch and add a correct changelog entry

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 6]
```diff
@@ -1263,7 +1296,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
       this.setA11ySelectionMessage(getA11ySelectionMessage.onRemove(poppedItem))
     }
 
-    this.setStateAndInvokeHandler('onSelectedChange', null, { value })
+    this.setStateAndInvokeHandler(['onSelectedChange'], null, { value })
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx:1299]: `onChage` may be a better alternative than `onSelectedChange` is it already exists as a concept in the html `select` element

File: packages/react/src/components/Dropdown/Dropdown.tsx [hunk 7]
```diff
@@ -1272,7 +1305,7 @@ class Dropdown extends AutoControlledComponent<WithAsProp<DropdownProps>, Dropdo
    * because we want to keep the event handling interface
    */
   setStateAndInvokeHandler = (
-    handlerName: keyof DropdownProps,
+    handlerNames: string[],
```
Comment (mnajdova) [inline packages/react/src/components/Dropdown/Dropdown.tsx:1308]: Why did we change `keyof DropdownProps` to `string`? Can't we make this array of keyof DropdownProps?

**Pull request closed without merging**